### PR TITLE
closes #1090: merge TypeException and TypingException

### LINKED ISSUE #1090: [FEATURE] Merge `TypeException` and `TypingException`
Currently, there are 2 exception classes related to internal type exceptions (+ `TypingInputException` for input problems). One of them is redundant.

### PR DESCRIPTION
**Pull request by konnov**
closes #1090. Merging `TypeException` and `TypingException`. Many files affected.

- [ ] ~Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala [hunk 1]
```diff
@@ -61,10 +61,12 @@ class CheckerExceptionAdapter @Inject() (sourceStore: SourceStore, changeListene
...
 
     case err: TypingInputException =>
-      NormalErrorMessage("Typing input error: " + err.getMessage)
+      // this is a normal message, as we know that the typing error is due to the user
+      NormalErrorMessage("%s: type input error: %s".format(findLoc(err.causeExprId), err.getMessage))
 
     case err: TypingException =>
-      FailureMessage("Type checker error: " + err.getMessage)
+      // this is a failure message, as we know that something type-related in apalache is broken
+      FailureMessage("%s: type error: %s".format(findLoc(err.causeExprId), err.getMessage))
```
Comment (shonfeder) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala:69]: Should the message reflect the comment? Perhaps something like:

```suggestion
      FailureMessage("%s: internal error in type checking: %s".format(findLoc(err.causeExprId), err.getMessage))
```

**Review by shonfeder — CHANGES_REQUESTED** [review]
Generally LGT, tho I have some questions about the messaging.

File: UNRELEASED.md [hunk 2]
```diff
@@ -13,3 +13,7 @@
 ### Bug fixes
 
 * Remove duplicate function indices when decoding symbolic states, fixes #962
+
+### Refactoring
+
+* Merge TypeException and TypingException, see #1090
```
[the remark this answers is not in the document]
Comment (konnov) [reply UNRELEASED.md:19]: Right. I was not sure either. But yes, it deserves a note on change in the message format.

[the remark this answers is not in the document]
Comment (konnov) [reply UNRELEASED.md]: Now it says:

> Change the format of type exceptions, see #1090

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala [hunk 3]
```diff
@@ -61,10 +61,12 @@ class CheckerExceptionAdapter @Inject() (sourceStore: SourceStore, changeListene
       NormalErrorMessage("%s: Input error (see the manual): %s".format(findLoc(err.causeExpr.ID), err.getMessage))
 
     case err: TypingInputException =>
-      NormalErrorMessage("Typing input error: " + err.getMessage)
+      // this is a normal message, as we know that the typing error is due to the user
+      NormalErrorMessage("%s: type input error: %s".format(findLoc(err.causeExprId), err.getMessage))
```
[the remark this answers is not in the document]
Comment (konnov) [reply tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala:65]: Nice that the tests have caught it. Fixed.

**Comment by konnov** [comment]
This PR is passing all checks now and ready for your final say :)

**Review by shonfeder — APPROVED** [review]

**Pull request merged**